src/components/Priority/Priority.js

```diff
@@ -91,45 +91,112 @@ const marksWithoutLabels = [
-    { "priorityIcon": "impact.svg", "priorityItemTitle": "Impact" },
-    { "priorityIcon": "effort.png", "priorityItemTitle": "Effort" },
+    { priorityIcon: 'urgency.svg', priorityItemTitle: 'Urgency' },
+    { priorityIcon: 'importance.svg', priorityItemTitle: 'Importance' },
+    { priorityIcon: 'impact.svg', priorityItemTitle: 'Impact' },
+    { priorityIcon: 'effort.png', priorityItemTitle: 'Effort' },
   ]
-  const priorityItems = priorityItemVars.map((priorityItem, index) =>
-    <div className="priority_item">
-      <Icon className='priority_item_icon' name={priorityItem.priorityIcon} size='small' />
-      <div className="priority_item_title">{priorityItem.priorityItemTitle}</div>
-      <Slider
-        defaultValue={50}
-        getAriaValueText={valuetext}
-        aria-labelledby="discrete-slider-custom"
-        step={25}
-        valueLabelDisplay="auto"
-        marks={index === 0 ? marksWithLabels : marksWithoutLabels}
-        classes={{
-          root: classes.root, // class name, e.g. `classes-nesting-root-x`
-          label: classes.label, // class name, e.g. `classes-nesting-label-x`
-          rail: classes.rail,
-          mark: classes.mark,
-          markLabel: classes.markLabel,
-          offset: classes.offset,
-          track: classes.track,
-        }}
-      />
-    </div>);
-
+  const [values, setValues] = useState({
+    Urgency: 50,
+    Importance: 50,
+    Impact: 50,
+    Effort: 50,
+  })
   const [openMyVote, setOpenMyVote] = useState(false)
+  const handleOnChange = sliderType => (e, value) => {
+    if (!openMyVote) return
+    if (values[sliderType] !== value) {
+      setValues({
+        ...values,
+        [sliderType]: value,
+      })
+    }
+  }
+  useEffect(() => {
+    if (!openMyVote) return
+    let goal_vote = {
+      goal_address: goalAddress,
+      urgency: values['Urgency'] / 100,
+      importance: values['Importance'] / 100,
+      impact: values['Impact'] / 100,
+      effort: values['Effort'] / 100,
+      agent_address: whoami.entry.address,
+      unix_timestamp: Date.now(),
+    }
+
+    const vote = votes.find(value => {
+      return value.agent_address === whoami.entry.address
+    })
+
+    if (votes.length > 0 && vote !== undefined) {
+      updateGoalVote(goal_vote, vote.address).then(value => {})
+    } else {
+      createGoalVote({ goal_vote }).then(value => {})
+    }
+  }, [values])
+  const priorityItems = priorityItemVars.map((priorityItem, index) => {
+    return (
+      <div key={index} className='priority_item'>
+        <Icon
+          className='priority_item_icon'
+          name={priorityItem.priorityIcon}
+          size='small'
+        />
+        <div className='priority_item_title'>
+          {priorityItem.priorityItemTitle}
+        </div>
+        <Slider
+          onChangeCommitted={handleOnChange(priorityItem.priorityItemTitle)}
+          defaultValue={averageValues[index]}
```

Comment:
I think we need to make `defaultValue` dependent on `openMyVote`. Only if we are looking at "aggregate" view should the values be the average.

Reply:
done